janus_core/calculations/md.py

```diff
@@ -360,20 +361,74 @@ def _optimize_structure(self) -> None:
                 self.logger.info("Minimizing at step %s", self.dyn.nsteps)
             optimize(self.struct, **self.minimize_kwargs)
 
+    @property
+    def _parameter_prefix(self) -> str:
+        """
+        Ensemble parameters for output files.
+
+        Returns
+        -------
+        str
+           Formatted temperature range if heating and target temp if running md.
+        """
+
+        temperature_prefix = ""
+        if self.temp_start and self.temp_end:
+            temperature_prefix = f"-T{self.temp_start}-T{self.temp_end}"
+
+        if self.steps > 0:
+            temperature_prefix = f"{temperature_prefix}-T{self.temp}"
+
+        return temperature_prefix
+
+    @property
+    def _final_file(self) -> str:
+        """
+        Final state file name.
+
+        Returns
+        -------
+        str
+           File name for final state.
+        """
+
+        if not self.restart_stem:
+            return f"{self.struct_name}-{self.ensemble}-T{self.temp}-final.xyz"
+        # respect the users choice
+        return f"{self.restart_stem}-T{self.temp}-final.xyz"
+
+    @property
+    def _restart_file(self) -> str:
+        """
+        Restart file name.
+
+        Returns
+        -------
+        str
+           File name for restart files.
+        """
+        step = self.offset + self.dyn.nsteps
+        if not self.restart_stem:
+            return f"{self.struct_name}-{self.ensemble}-T{self.temp}-res-{step}.xyz"
```

Comment:
Can we not use `self.file_prefix` here (and in `_final_file`)? We seem to use `self.struct_name}-{self.ensemble}` etc. in a few difference places, when they ought to be handled together if possible.

Reply:
I think we can, simply the issue comes from the parameters potentially containing the temperature range from heating which only some files require.

I think then the file_prefix should be whatever is in common to all which is ```{self.struct_name}-{self.ensemble}``` only

Reply:
This is done

But I had to introduce a local for the stats and traj as their behaviour depends on whether ```file_prefix``` is set by the user or not. Which is lost information once ```file_prefix``` gets set if it was not. 

The difference in behaviour is that in the override case stats and traj's file name should not contain the temperature/ heating ranges/ pressure etc. (according to tests)